### PR TITLE
assigned flight calendar fix

### LINKED ISSUE #428: Availability on assigned flights may not be changed.
When a user is assigned to a flight he should not be able to modify the availability

### PR DESCRIPTION
**Pull request by hacknstack**
closes #428 
basically a rework of #342 in the new VM environement

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt [hunk 1]
```diff
@@ -79,12 +81,22 @@ class CalendarViewModel(
    */
   private suspend fun refreshUserAndCalendars() {
     _user.value = userTable.get(uid, this::onError)!!
-    val currentAvCal = AvailabilityCalendar(userTable.retrieveAvailabilities(uid, this::onError))
+    var currentAvCal = AvailabilityCalendar(userTable.retrieveAvailabilities(uid, this::onError))
+    val flights = userTable.retrieveAssignedFlights(flightTable, uid, this::onError)
+    flights.forEach { flight ->
+      if (flight !is FinishedFlight) {
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt:87]: status should also be set to `assigned` for any flight when affected not just for a `FinishedFlight`

**Review by RAFInery — CHANGES_REQUESTED** [review]
It would be nicer to update and persist the assigned status of the availability upon affection.  Like this it does not have to be checked on every load.

File: app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt (hunk 1 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt:87]: It's set to assigned for every flight that isn't finished

**Review by RAFInery — APPROVED** [review]
LGTM

**Pull request merged**